### PR TITLE
Various IE 1.1 fixes

### PR DESCRIPTION
**Pull request by nickjbenson**
This PR:

- Addresses issue #738 (Pressing X should not dismiss IE errors forever)
- Addresses issue #754 (Auto detect rigid hands and add a warning in the inspector) 
- Addresses issue #749 (Slider discretization is off-by-1)
- Addresses issue #765 (Clarify InteractionVRController device string comment)
- Addresses issue #767 (Add a warning when a slider's value is accessed prematurely)
- Merges in some changes and DevX improvements from Particles to UnityModules for `InteractionButton` and `InteractionSlider`, particularly:
  - InteractionButtons can now be modeled based on where they relax to instead of to where they depress
  - InteractionSliders have more detailed gizmos describing their limits
  - InteractionSliders can have RectTransforms themselves, and these RectTransforms "collide" against their parents' RectTransforms in the editor for defining the slider's movement range (addresses #766)
  - (a change from Alex) InteractionSliders now explicitly set whether they are two-dimensional or one-dimensional (and whether they are horizontal or vertical)

Roughly speaking, this PR and #771 represent the MVP for Interaction Engine v1.1.

To Test:
 - [ ] Verify pressing X does not dismiss IE physics/timestep errors forever
 - [ ] Verify that having RigidHands in your scene shows a warning in the Interaction Manager inspector
 - [ ] Check that the tooltip for horizontalSteps and verticalSteps is clear for InteractionSliders
 - [ ] Confirm the device joystick tokens property on InteractionControllers is clear
 - [ ] Mess around with InteractionButtons and InteractionSliders and confirm the new features work properly

### CODE REVIEW

File: Assets/LeapMotion/Modules/InteractionEngine/Scripts/UI/InteractionButton.cs [hunk 1]
```diff
@@ -21,8 +24,25 @@ namespace Leap.Unity.Interaction {
   ///</summary>
   public class InteractionButton : InteractionBehaviour {
 
+    [Header("UI Control")]
+    [Tooltip("When set to false, this UI control will not be functional. Use this instead "
+           + "of disabling the component itself when you want to disable the user's "
+           + "ability to affect this UI control.")]
+    public bool controlEnabled = true;
+    public void SetControlEnabled(bool enableControl) {
```
Comment (Amarcolina) [inline Assets/LeapMotion/Modules/InteractionEngine/Scripts/UI/InteractionButton.cs]: Any reason to have a public variable _and_ a public method that can set/reset it?

File: Assets/LeapMotion/Modules/InteractionEngine/Scripts/UI/InteractionSlider.cs [hunk 2]
```diff
@@ -103,25 +133,101 @@ public class FloatEvent : UnityEvent<float> { }
 
     private bool _started = false;
 
+    public bool hackModeOn = false;
```
Comment (Amarcolina) [inline Assets/LeapMotion/Modules/InteractionEngine/Scripts/UI/InteractionSlider.cs]: Don't think we should include 'hack mode' in unity modules :D

File: Assets/LeapMotion/Modules/InteractionEngine/Scripts/UI/InteractionButton.cs (hunk 1 above)
Comment (nickjbenson) [inline Assets/LeapMotion/Modules/InteractionEngine/Scripts/UI/InteractionButton.cs]: probably?

File: Assets/LeapMotion/Modules/InteractionEngine/Scripts/UI/InteractionSlider.cs (hunk 2 above)
Comment (nickjbenson) [inline Assets/LeapMotion/Modules/InteractionEngine/Scripts/UI/InteractionSlider.cs]: fixed

File: Assets/LeapMotion/Modules/InteractionEngine/Scripts/UI/InteractionButton.cs (hunk 1 above)
Comment (nickjbenson) [inline Assets/LeapMotion/Modules/InteractionEngine/Scripts/UI/InteractionButton.cs]: k replaced it with the public property / private serialized variable pattern

**Comment by nickjbenson** [comment]
Pushed a fix to move `specifyConditionalDrawing` to OnEnable. John has some specifyDecorator code still in OnInspectorGUI though, should that also be moved?

**Comment by Amarcolina** [comment]
Yeah, all 'specify' code should always be done in onEnable.

**Comment by nickjbenson** [comment]
@Amarcolina OK, all `specify` code has been migrated to OnEnable for Interaction Sliders

**Pull request merged**